website/docs/guides/upgrade.html.markdown

```diff
@@ -0,0 +1,159 @@
+---
+layout: "nsxt"
+page_title: "VMware NSX Terraform Provider NSX upgrade support"
+description: |-
+  VMware NSX Terraform Provider NSX upgrade support
+---
+
+# NSX Provider for NSX Upgrade
+
+NSX Terraform Provider offers support for [upgrading NSX components](https://docs.vmware.com/en/VMware-NSX/4.1/upgrade/GUID-E04242D7-EF09-4601-8906-3FA77FBB06BD.html).
+
+The provider support consists of several resources and data sources which can perform a complete upgrade cycle.
+The NSX Terraform provider support upgrades for NSX on ESX hosts, NSX Edge appliances, and NSX manager appliances.
+
```

Comment:
Lets emphasize that upgrade configuration and state should be dedicated to upgrade only, and not shared with regular terraform config and state
Also do we want to recommend that state is cleared post successful upgrade?

Reply:
Done - I added a note at the beginning of the doc, about state management for upgrade.